### PR TITLE
Modelrun category

### PR DESCRIPTION
**Pull request by zsxing99**
expanded model category: public, owner, base
enabled "put" on models(publish models)

### CODE REVIEW

File: npsat_manager/views.py
Comment (nickrsan) [inline npsat_manager/views.py:114]: Let's get a test committed for this function to make sure that the permissions behave - seems like a function we might modify in the future where we could accidentally expose all model runs to everyone. Looks good otherwise

File: npsat_manager/views.py
Comment (nickrsan) [inline npsat_manager/views.py:198]: not sure we need to use Q objects in here, or that we need the select_related. Should just be able to do the direct filtering. 

Also, looks like we're returning all results every time any new results are requested? I think it'd be good if we had it structured so that there was an endpoint for all objects, but also to get a single set of results (so that a user with 20 model runs can run a new model, and we only need to retrieve the new results from the client?) - not a big deal, but an enhancement to consider.

**Review by nickrsan — CHANGES_REQUESTED** [review]
Let's make at least some of the changes mentioned inline before doing a merge here

File: npsat_manager/views.py
Comment (zsxing99) [reply npsat_manager/views.py:198]: This will first filter out all results available to a user. The model result doesn't have any field indicating whom it belongs to, so by including `model`, we can find the model creator who owns the results. This queryset basically filters results that are owned by the user, are public, or are base model results.

Currently, this endpoint is only used when querying single model results, but I can make it support multiple models results query.

File: npsat_manager/views.py
Comment (zsxing99) [reply npsat_manager/views.py:114]: Permissions should be good. It will first authenticate the user, then gets the user which kinds of models based on the params.

I will push a test case with the next commit.

File: npsat_manager/views.py
Comment (nickrsan) [reply npsat_manager/views.py:198]: Totally, we definitely need to traverse the relationship and I made a mistake in saying we don't need the Q objects - I forgot that we'd need them for an OR query like this. That said, you shouldn't need the `select_related` call - that might trigger an extra DB query we don't need. It should work fine without it, but let me know if it doesn't.

Comment (zsxing99) [reply npsat_manager/views.py:197]: Oh, didn't realize that we can use `model` directly without `select_related`. I must have messed with ORM and raw SQL. Thanks!

**Review by nickrsan — APPROVED** [review]

**Comment by nickrsan** [comment]
Looks like a great set of changes! I think we're good to go and if we notice anything, we can open a new pull request - but I think this is in great shape. Thanks Zesheng!

**Pull request merged**
